crates/wasi-common/yanix/src/file.rs

```diff
@@ -230,20 +233,26 @@ pub unsafe fn fstatat<P: AsRef<OsStr>>(dirfd: RawFd, path: P, flags: AtFlag) ->
 pub unsafe fn fstat(fd: RawFd) -> Result<libc::stat> {
     use std::mem::MaybeUninit;
     let mut filestat = MaybeUninit::<libc::stat>::uninit();
-    Error::from_result(libc::fstat(fd, filestat.as_mut_ptr()))?;
+    from_result(libc::fstat(fd, filestat.as_mut_ptr()))?;
     Ok(filestat.assume_init())
 }
 
 /// `fionread()` function, equivalent to `ioctl(fd, FIONREAD, *bytes)`.
 pub unsafe fn fionread(fd: RawFd) -> Result<u32> {
     let mut nread: libc::c_int = 0;
-    Error::from_result(libc::ioctl(fd, libc::FIONREAD, &mut nread as *mut _))?;
-    Ok(nread.try_into()?)
+    from_result(libc::ioctl(fd, libc::FIONREAD, &mut nread as *mut _))?;
+    // FIONREAD returns a non-negative int if it doesn't fail, or it'll fit in a u32 if it does.
+    //
+    // For the future, if we want to be super cautious and avoid assuming int is 32-bit, we could
+    // widen fionread's return type here, since the one place that calls it wants a u64 anyway.
+    Ok(nread.try_into().unwrap())
```

Comment:
FWIW I think ideally we would use `nread.try_into()?` here. I agree with the comment here but "unwrap" is pretty scary when reading this sort of code and feels like something that would best be entirely blanket avoided. If the `?`-conversion works it's also even more ergonomic than `nread.try_into().unwrap()`!

Basically I like that there's a comment here to explain why the unwrap is ok, and I don't disagree with it, but I think it would be better idiomatically to use `?` if we can to propagate errors to consistently "never use unwrap" anywhere in wasi-common. (in theory at least). If the error handling is more verbose than the literal 1-character `?` symbol though then this is fine.

Reply:
@sunfishcode has some counterpoints to this thinking [on Zulip](https://bytecodealliance.zulipchat.com/#narrow/stream/217126-wasmtime/topic/never.20use.20unwrap.3F), so I don't think that we should remove the `.unwrap()` for now. Let's leave it here and handle the bug report if it ever comes in!

Reply:
FWIW, both of you guys make really good points. So in order to strike a middle ground, I've made sure to leave comments at every occurrence of `unwrap`.

OK, so ultimately let's stick with `unwrap`s for now, until we get some bug reports. :+1: